stl/inc/xutility

```diff
@@ -2215,19 +2215,14 @@ namespace ranges {
 
     // CONCEPT ranges::range
     template <class _Rng>
-    concept _Range_impl = requires(_Rng&& __r) {
-        _RANGES begin(static_cast<_Rng&&>(__r));
-        _RANGES end(static_cast<_Rng&&>(__r));
+    concept range = requires(_Rng& __r) {
+        _RANGES begin(__r);
+        _RANGES end(__r);
     };
 
+    // CONCEPT ranges::safe_range
     template <class _Rng>
-    concept range = _Range_impl<_Rng&>;
-
-    // CONCEPT ranges::_Forwarding_range
-    // clang-format off
-    template <class _Rng>
-    concept _Forwarding_range = range<_Rng> && _Range_impl<_Rng>;
-    // clang-format on
+    concept safe_range = range<_Rng>&& _Safe_range_like<_Rng>;

```

Comment:
This lack of space before `&&` seems like a clang-format bug. You might have to use // clang-format off/on around this line.

Reply:
Yeah I thought about that but decided against it due to the larger distortion of clang-format switches versus the missing space. I would wait for a maintainers verdict on that.

BTW sharp eyes :+1:

Reply:
Apart from this everything looks good ;)